workflow/util/util_test.go

```diff
@@ -1367,6 +1367,37 @@ func TestFormulateRetryWorkflow(t *testing.T) {
 			}
 		}
 	})
+
+	t.Run("Nested DAG with Non-group Node Selected, NO restartSuccessful", func(t *testing.T) {
```

Comment:
this should be worded the same way as the other test cases
```suggestion
	t.Run("Retry failed workflow with only nodeFieldSelector", func(t *testing.T) {
```